rpcs3/Emu/Cell/lv2/sys_net.cpp

```diff
@@ -1201,7 +1442,7 @@ error_code sys_net_bnet_recvfrom(ppu_thread& ppu, s32 s, vm::ptr<void> buf, u32
 	::sockaddr_storage native_addr;
 	::socklen_t native_addrlen = sizeof(native_addr);
 	sys_net_error result{};
-	std::string _buf;
+	std::vector<u8> _buf;
```

Comment:
std::vector may return an invalid pointer on .data() when it is empty, so passing over such pointer to operating systems may result in an error. its not an issue with std::string.

Reply:
We should go with Silent's suggestion of a `basic_string<u8>`, then.

Reply:
recvfrom works fine if buf is nullptr and len is 0 which would be our case here.